toqito/state_props/is_separable.py

```diff
@@ -22,14 +20,150 @@
+        subsystem dimensions :math:`d_A, d_B` is processed or inferred.
+
+    2.  **Trivial Cases for Separability**:
+
+        - If either subsystem dimension :math:`d_A` or :math:`d_B` is 1
+          (i.e., :code:`min_dim_val == 1`), the state is always separable.
+
+    3.  **Pure State Check (Schmidt Rank)**:
+
+        - If the input state has rank 1 (i.e., it's a pure state), its Schmidt rank is computed.
+          A pure state is separable if and only if its Schmidt rank is 1 :cite:`WikiScmidtDecomp`.
+        .. note::
+           QETLAB also considers a more general Operator Schmidt Rank condition
+           from :cite:`Cariello_2013_Weak_irreducible` for weak irreducible
+           matrices. This is not explicitly separated in this function but might be
+           covered if such matrices are rank 1. This is a candidate for a
+           future enhancement (see Issue #XYZ).
+
+
+    4.  **Gurvits-Barnum Separable Ball**:
+
+        - Checks if the state lies within the "separable ball" around the
+          maximally mixed state, as defined by Gurvits and Barnum
+          :cite:`Gurvits_2002_Ball`. States within this ball are guaranteed to be
+          separable.
+
+    5.  **PPT Criterion (Peres-Horodecki)**
+        :cite:`Peres_1996_Separability`,
+        :cite:`Horodecki_1996_PPT_small_dimensions`:
+
+        - The Positive Partial Transpose (PPT) criterion is a necessary condition
+          for separability.
+        - If the state is NPT (Not PPT), it is definitively entangled.
+        - If the state is PPT and the total dimension :math:`d_A d_B \le 6`,
+          then PPT is also a *sufficient* condition for separability
+          :cite:`Horodecki_1996_PPT_small_dimensions`.
+
+
+    6.  **3x3 Rank-4 PPT N&S Check (Plücker Coordinates / Breuer / Chen & Djokovic)**:
+
+        - For 3x3 systems, if a PPT state has rank 4, there are known
+          necessary and sufficient conditions for separability. These are often
+          related to the vanishing of the "Chow form" or determinants of
+          matrices constructed from Plücker coordinates of the state's range
+          (e.g., :cite:`Breuer_2006_Optimal`, :cite:`Chen_2013_MultipartiteRank4`).
+          The implementation checks if a specific determinant (:code:`F_det_val`)
+          is close to zero.
+
+    7.  **Operational Criteria for Low-Rank PPT States (Horodecki et al. 2000)** :cite:`Horodecki_2000_PPT_low_rank`:
+
+        For PPT states (especially when :math:`d_A d_B > 6`):
+
+        - If :math:`\text{rank}(\rho) \le \max(d_A, d_B)`, the state is
+          separable.
+        - If :math:`\text{rank}(\rho) + \text{rank}(\rho^{T_A}) \le 2 d_A d_B - d_A - d_B + 2`, the state is separable.
+
+    8.  **Reduction Criterion (Horodecki & Horodecki 1999)** :cite:`Horodecki_1998_Reduction`:
+
+        - If :math:`I_A \otimes \rho_B - \rho \not\succeq 0`
+          or :math:`\rho_A \otimes I_B - \rho \not\succeq 0` (where
```

Comment:
Just double checking, is this supposed to be 'not greater than' or 'being majorized by'?